wgsl/index.bs

```diff
@@ -11737,8 +10999,861 @@ When calling a built-in function, all arguments to the function are evaluated
+@const @must_use fn vec4<T>(v1 : vec3<T>, e1 : T) -> vec4<T>
+@const @must_use fn vec4(v1 : vec3<T>, e1 : T) -> vec4<T>
+</xmp>
+  <tr><td>Parameterization
+      <td>`T` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a four-component [=vector=] with `v1.x`, `v1.y`, `v1.z`, and `e1` as components.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>
+@const @must_use fn vec4<T>(e1 : T, v1 : vec3<T>) -> vec4<T>
+@const @must_use fn vec4(e1 : T, v1 : vec3<T>) -> vec4<T>
+</xmp>
+  <tr><td>Parameterization
+      <td>`T` is [=scalar=]
+  <tr><td>Description
+      <td>[=Component-wise=] construction of a four-component [=vector=] with `e1`, `v1.x`, `v1.y`, and `v1.z` as components.
+</table>
+
+## Value Reinterpretation Built-in Functions ## {#value-reinterp-builtin-functions}
+
+### `bitcast` ### {#bitcast-builtin}
+
+A `bitcast` built-in function is used to reinterpret the bit representation of
+a value in one type as a value in another type.
+
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>@const @must_use fn bitcast<T>(e : T) -> T</xmp>
+  <tr><td>Parameterization
+      <td>`T` is a [=type/concrete=] [=numeric scalar=] or [=type/concrete=] [=numeric vector=]
+  <tr><td>Description
+      <td>Identity transform.<br>
+      [=Component-wise=] when `T` is a [=vector=].<br>
+      The result is `e`.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>@const @must_use fn bitcast<T>(e : S) -> T</xmp>
+  <tr><td>Parameterization
+      <td>`S` is i32, u32, or f32<br>
+      `T` is not `S` and is i32, u32, or f32
+  <tr><td>Description
+      <td>Reinterpretation of bits as `T`.<br>
+      The result is the reintepretation of bits in `e` as a `T` value.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>@const @must_use fn bitcast<vecN<T>>(e : vecN<S>) -> T</xmp>
+  <tr><td>Parameterization
+      <td>`S` is i32, u32, or f32<br>
+      `T` is not `S` and is i32, u32, or f32
+  <tr><td>Description
+      <td>[=Component-wise=] reinterpretation of bits as `T`.<br>
+      The result is the reintepretation of bits in `e` as a `vecN<T>` value.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>@const @must_use fn bitcast<T>(e : vec2<16>) -> T</xmp>
+  <tr><td>Parameterization
+      <td>`T` is i32, u32, or f32
+  <tr><td>Description
+      <td>[=Component-wise=] reinterpretation of bits as `T`.<br>
+      The result is the reintepretation of the 32 bits in `e` as a `T` value, following the internal layout rules.
+</table>
+<table class='data builtin'>
+  <tr><td style="width:10%">Overload
+      <td><xmp highlight=rust>@const @must_use fn bitcast<T>(e : vec4<f16>) -> T</xmp>
+  <tr><td>Parameterization
+      <td>`T` is i32, u32, or f32
```

Comment:
Should be `S` is i32, u32, or f32, `T` is vec2<`S`>

Reply:
Left `T` as is and explicitly used `vec2<T>` in the overload to match other cases.